src/videojs-http-streaming.js

```diff
@@ -119,6 +135,46 @@ Hls.canPlaySource = function() {
                           'your player\'s techOrder.');
 };
 
+const emeOptions = (keySystemOptions, videoPlaylist, audioPlaylist) => {
+  if (!keySystemOptions) {
+    return keySystemOptions;
+  }
+
+  // upsert the content types based on the selected playlist
+  const keySystemContentTypes = {};
+
+  for (let keySystem in keySystemOptions) {
+    keySystemContentTypes[keySystem] = {
+      audioContentType: `audio/mp4; codecs="${audioPlaylist.attributes.CODECS}"`,
+      videoContentType: `video/mp4; codecs="${videoPlaylist.attributes.CODECS}"`
+    };
+
+    // videojs-contrib-eme accepts the option of specifying: 'com.some.cdm': 'url'
+    // so we need to prevent overwriting the URL entirely
+    if (typeof keySystemOptions[keySystem] === 'string') {
+      keySystemContentTypes[keySystem].url = keySystemOptions[keySystem];
+    }
+  }
+
+  return {
+    keySystems: videojs.mergeOptions(keySystemOptions, keySystemContentTypes)
+  };
+};
+
+const setupEmeOptions = (hlsHandler) => {
```

Comment:
probably would be better to use `vhsHandler` naming

Reply:
just realized, are we not making name changes for `HlsHandler` my hls->vhs name change suggestions probably shouldn't be done then